_episodes_rmd/03-r-data-analysis.Rmd

```diff
@@ -33,258 +32,315 @@ knitr_fig_path("03-")
 
 ```{r GapLifeExpByYear}
-gapminder_data %>%
+smoking %>%
   group_by(year) %>%
-  summarize(average=mean(lifeExp))
+  summarize(average=mean(smoke_pct))
 ```
 
-The `group_by()` function expects you to pass in the name of a column (or multiple columns separated by comma) in your data. 
+The `group_by()` function expects you to pass in the name of a column (or multiple columns separated by commas) in your data. 
 
 Note that you might get a message about the summarize function regrouping the output by 'year'. This simply indicates what the function is grouping by. 
 
+> ## Pro tip: Counting
+> 
+> Often we want to count how many of each category we have in our dataset. 
+> To do this, we can use the function `n()` within `summarize()`. 
+> For instance, here we count the number of countries in each continent for which we have data in 2019:
+> 
+> ```{r}
+> smoking %>% 
+>   filter(year == 2019) %>% 
+>   group_by(continent) %>% 
+>   summarize(count = n())
+> ```
+> 
+{: .callout}
+
+
 > ## Grouping the data
 >
-> Try calculating the average life expectancy by continent.
+> Try calculating the average percent of smokers by continent.
 >
 > > ## Solution
 > >
 > > ```{r GroupByChallenge1}
-> > gapminder_data %>%
+> > smoking %>%
 > > group_by(continent) %>%
-> > summarize(average=mean(lifeExp))
+> > summarize(avg_smoke_pct=mean(smoke_pct))
 > > ```
 > > {: .source}
 > >
-> > By combining `group_by()` and `summarize()` we are able to calculate the mean life expectancy by continent.
+> > By combining `group_by()` and `summarize()` we are able to calculate the mean percent of smokers by continent.
 > {: .solution}
 {: .challenge}
 
-You can also create more than one new column when you call `summarize()`. To do so, you must separate your columns with a comma. Building on the code from the last exercise, let's add a new column that calculates the minimum life expectancy for each continent. 
 
-```{r GapLifeExpMinMax}
-gapminder_data %>%
-  group_by(continent) %>%
-  summarize(average=mean(lifeExp), min=min(lifeExp))
-```
+> ## Bonus: plotting review
+>
+> Make a line plot of year vs. average percent of smokers to be able to more easily visualize the relationship between these two variables. 
+>
+> > ## Solution
+> >
+> > ```{r}
+> > smoking %>%
+> >   group_by(year) %>%
+> >   summarize(average=mean(smoke_pct)) %>% 
+> >   ggplot(aes(x = year, y = average)) +
+> >   geom_line()
+> > ```
+> > {: .source}
+> >
+> > You can pipe your summarized data right to ggplot! 
+> {: .solution}
+{: .challenge}
 
```

Comment:
Okay, I know we're doing a lot in here, but I feel like these are some of THE most important skills to teach. What do you think of an example where we group by two variables? Maybe year and continent? That would be manageable output, I think?